webapp/src/components/tutorial/TutorialContainer.tsx

```diff
@@ -106,9 +109,77 @@ export function TutorialContainer(props: TutorialContainerProps) {
     const markdown = steps[visibleStep].headerContentMd;
     const hintMarkdown = steps[visibleStep].hintContentMd;
 
+    const validateTutorialStep = async () => {
+        const globalValidators = tutorialOptions.globalValidationConfig?.validators
+                                    ?? PopulateValidatorCache(tutorialOptions.globalValidationConfig);
+        const localValidators = currentStepInfo.localValidationConfig?.validators
+                                    ?? PopulateValidatorCache(currentStepInfo.localValidationConfig);
+
+        let validators: pxt.Map<CodeValidator> = {}
+        if(localValidators) {
+            Object.entries(localValidators).forEach(v => validators[v[0]] = v[1]);
+        }
+        if (globalValidators) {
```

Comment:
could we just declare validators as 

```typescript
const validators: pxt.Map<CodeValidator> = {
    ...(globalValidators ?? {}),
    ...(localValidators ?? {}),
};
```

? should get the behavior you want of localValidators overriding globals and maybe a little easier to read.

(will defer to you though as they'll ultimately do the same thing~)

Reply:
Much cleaner, thanks.